plan/refiner.go

```diff
@@ -128,6 +128,58 @@ func removeAccessConditions(conditions, accessConds []expression.Expression) []e
 	return conditions
 }
 
+func judgeIndexExpression(arg expression.Expression, indexColumns []*model.IndexColumn, pKName model.CIStr) bool {
+	_, ok := arg.(*expression.Constant)
+	if ok {
+		return true
+	}
+	if c, ok := arg.(*expression.Column); ok {
+		if pKName.L == c.ColName.L {
+			return true
+		}
+		for _, col := range indexColumns {
+			if col.Name.L == c.ColName.L && col.Length == types.UnspecifiedLength {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func judgeIndexCondition(condition expression.Expression, indexColumns []*model.IndexColumn, pKName model.CIStr) bool {
```

Comment:
Use `check` instead of `judge`.
And we should just use one function, do recursive call, because an argument may be a scalar function.